test/integration/features/profile.feature

```diff
@@ -2,104 +2,251 @@
-    When executing "minishift profile list" succeeds
-    Then stdout should contain
-    """
-    - minishift	Running		(Active)
-    """
-
-Scenario: User should be able to delete profile 'foo'
-    When executing "minishift profile delete foo --force" succeeds
-    Then stdout should contain
-    """
-    Profile 'foo' deleted successfully
-    """
-
-Scenario: User can not delete default profile 'minishift'
-    When executing "minishift profile delete minishift --force"
-    Then exitcode should equal "1"
-     And stderr should contain
+  Scenario: As user, I can start Minishift with default profile 'minishift'
+    Given Profile "minishift" has state "Does Not Exist"
+     And executing "minishift profile set minishift" succeeds
+     Then stdout should contain
      """
-     Default profile 'minishift' can not be deleted
+     Profile 'minishift' set as active profile
      """
+     When executing "minishift start" succeeds
+      Then Profile "minishift" should have state "Running"
+       And executing "minishift profile list" succeeds
+       And stdout should match "-\s*minishift\s*Running\s*\(Active\)"
 
-Scenario: Deleting Minishift
-   Given Minishift has state "Running"
-    When executing "minishift delete --force" succeeds
-    Then Minishift should have state "Does Not Exist"
-    When executing "minishift ip"
-    Then exitcode should equal "1"
+  Scenario: As user, I can get internal docker registry address of default profile 'minishift'
+    Given Profile "minishift" has state "Running"
+     And executing "minishift profile set minishift" succeeds
+     Then stdout should match "Profile\s*\'minishift\'\s*set\s*as\s*active\s*profile"
+     When executing "minishift openshift registry" succeeds
+      Then stdout should be valid IP with port number
 
-Scenario: User should be able to switch between non existing profiles
-    When executing "minishift profile list" succeeds
-    Then stdout should contain
-    """
-    - minishift	Does Not Exist	(Active)
-    """
-    When executing "minishift profile set abc" succeeds
-      Then stdout should contain
-      """
-      Profile 'abc' set as active profile.
+  Scenario: As user, I can apply independent settings in profile 'foo'
+    Given Profile "foo" has state "Does Not Exist"
+     And executing "minishift profile set foo" succeeds
+     Then stdout should match "Profile\s*\'foo\'\s*set\s*as\s*active\s*profile"
+      When executing "minishift config set memory 4096" succeeds
+       And executing "minishift config set disk-size 30g" succeeds
+       And executing "minishift config set cpus 4" succeeds
+       And executing "minishift addons enable registry-route" succeeds
+         
+  @minishift-only
+  Scenario: As user, I can check the config settings and add-ons list that should not be changed in profile 'minishift' 
+    Given Profile "foo" has state "Does Not Exist"
+     And executing "minishift profile set foo" succeeds
```

Comment:
`Scenario: As user, I can check the config settings and add-ons list that should not be changed in profile 'minishift' ` but you are not really checking the `minishift` profile, but the `foo` profile instead.

Reply:
There is some missing text in the scenario description imo. 

This should be something like 
```
Scenario: As user, I can check the config settings and add-ons list that should not be changed in profile 'minishift' from active profile 'foo'
```